server/controllers/donor.controller.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const { Donor, Event, Gift, User, Profile } = require('../../config/db');
+const { messages } = require('./../helper');
+
+module.exports = {
+  create(req, res) {
+    let assignDonor = Object.assign({}, req.body, {gift_id: req.params.gift_id},
+      {user_id: req.params.id});
+    Donor.findOne({
+      where: {
+        user_id: req.params.id,
+        gift_id: req.params.gift_id
+      }
+    })
+      .then(donor => {
+        !donor &&
+        Gift.findOne({
+          where: {
+            id: req.params.gift_id,
+            is_available: true
+          }
+        }).then(gift => {
+          !gift && res.status(400).json({'message': messages.badRequest});
+          gift.dataValues.status == 'hasOneDonor' &&
+          gift.update({
+            is_available: false
+          });
+          Donor.create(assignDonor)
+            .then(donor => res.status(201).send(donor));
+        })
+          .catch(error => res.status(400).send(error)) ||
+            res.status(400).json({'message': messages.badRequest});
+      })
+        .catch(error => res.status(400).send(error)) ||
```

Comment:
i rather would leave send error...because catch use for error and don't use bad request - it's not error.